### ISSUE TITLE
Add support for object valued properties like navigation.attitude

### ISSUE THREAD

**Issue by tkurki** [SignalK/calibration#6]
The plugin can currently operate only on number valued properties, not on objects like `{ "yaw": null, "pitch": -0.009, "roll": -0.1584 }`.

**Comment by chrischanHH** [comment]
Hi tkurki, I try to implement the calibration plugin and the needed linear-interpolatorin Signalk but I do not get  data in the log. Only I can see in the server log that I use the plagin and they settings. Have I add some code to the raspi to run the plugin or should it run out 'of the box'. I want calibrat the simple temperature for example. I use the news version of SK and @signalk/calibration
Can you help me?
(-: Chrischan

**Comment by tkurki** [comment]
Please join signalk Slack at signalk.org, that's the best way to get support. Commenting on a totally unrelated issue is not.

**Comment by chrischanHH** [comment]
Okay thanks, I try my very best 

Von meinem iPhone gesendet

> Am 06.11.2022 um 19:58 schrieb Teppo Kurki ***@***.***>:
> 
> Commenting on a totally unrelated issue is not.

**Comment by macjl** [comment]
I wonder if the issue is with the navigation.attitude path design itself rather than the plugin.
Unlike navigation.position where lat/lon are meaningless when separated, each attitude component (yaw, pitch, roll) represents an independent rotation axis that can be measured and calibrated independently.
Proposal: Restructure as three separate paths:
```
navigation.attitude.yaw
navigation.attitude.pitch
navigation.attitude.roll
```
Benefits:
	∙	Works immediately with existing calibration plugin
	∙	Better granularity for subscriptions/updates
	∙	Simpler for consumers needing specific axes
	∙	Aligns with standard Signal K property patterns
This would be a breaking change, but navigation.attitude and navigation.position appear to be the only object-valued properties needing calibration in the core schema.
Thoughts?

**Comment by tkurki** [comment]
I agree. Looking for volunteer to comb through the current implementation and importabt plugins.

**Comment by macjl** [comment]
I’ve just created [signalk-attitude-converter](https://github.com/macjl/signalk-attitude-converter) as a transition bridge while the ecosystem moves to this model. To be honest, the AI did it !

For now, the plugin isn’t published to npm. I’m testing it on my own boat.

The goal is to expose attitude as independent scalar values (navigation.attitude.yaw / pitch / roll), while maintaining compatibility with existing producers and consumers that still use the object-valued navigation.attitude.

The target is for scalar attitude values to become the norm; this plugin helps manage the migration cost while plugins and tools gradually adapt.

Happy to help review impacted plugins or migration paths if useful.
